.github/workflows/codeql.yml

```diff
@@ -26,6 +26,10 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
+    - name: Setup Go
+      uses: actions/setup-go@v4
+      with:
+        go-version-file: './go.mod'
```

Comment:
Wonder if this is required as it seems to be the "default" location for this file?

Reply:
it tells the setup-go action to setup the go version in the go.mod. Without it, it would just use whatever version is already there or install the latest. https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file